src/parse/grammar/gast/gast_public.ts

```diff
@@ -177,34 +177,59 @@ export class Terminal implements IProductionWithOccurrence {
     public idx: number = 1
 
     constructor(options: { terminalType: TokenType; idx?: number }) {
-        assign(this, options)
+        assign(this, pick(options, v => v !== undefined))
     }
 
     accept(visitor: IGASTVisitor): void {
         visitor.visit(this)
     }
 }
 
+export interface ISerializedBasic extends ISerializedGast {
+    type:
```

Comment:
1. We already use serialization for drawing grammar diagrams
    How did we do that without adding a "type" property?
    or was the type property already there, just not documented in the types?

2. What about the "WithSeparator" types?